### PR TITLE
DDCNOS-1642: valuation date change for 2017

### PR DESCRIPTION
**Pull request by shikkhasingh**

### CODE REVIEW

File: app/models/propertyDetailsModels.scala [hunk 1]
```diff
@@ -85,8 +124,26 @@ object PropertyDetailsRevalued {
...
+case object IsOwnedBefore2017 extends OwnedBeforePolicyYear
+
+case object NotOwnedBeforePolicyYear extends OwnedBeforePolicyYear
+
+case class PropertyDetailsOwnedBefore(isOwnedBeforePolicyYear: Option[Boolean] = None,
+                                      ownedBeforePolicyYearValue: Option[BigDecimal] = None) {
+
+  def policyYear(periodKey: Int) = isOwnedBeforePolicyYear match {
+    case Some(true) => periodKey match {
+      case p if p >= 2018 => IsOwnedBefore2017
```
Comment (daniel-manning) [inline app/models/propertyDetailsModels.scala]: Name doesnt match logic

File: app/models/propertyDetailsModels.scala [hunk 2]
```diff
@@ -85,8 +124,26 @@ object PropertyDetailsRevalued {
...
+      case p if p >= 2013 && p < 2018 => IsOwnedBefore2012
```
Comment (daniel-manning) [inline app/models/propertyDetailsModels.scala]: Same

File: app/utils/PropertyDetailsUtils.scala [hunk 3]
```diff
@@ -23,6 +23,9 @@ import uk.gov.hmrc.http.InternalServerException
 
 object PropertyDetailsUtils extends ReliefConstants {
 
+  val _2012ValuationPolicyDate = Some(new LocalDate("2012-04-01"))
+  val _2017ValuationPolicyDate = Some(new LocalDate("2017-04-01"))
```
Comment (daniel-manning) [inline app/utils/PropertyDetailsUtils.scala]: these can be worked out from the period you fall into

File: app/models/propertyDetailsModels.scala [hunk 4]
```diff
@@ -85,8 +124,26 @@ object PropertyDetailsRevalued {
   implicit val formats = Json.format[PropertyDetailsRevalued]
 }
 
-case class PropertyDetailsOwnedBefore(isOwnedBefore2012: Option[Boolean] = None,
-                                      ownedBefore2012Value: Option[BigDecimal] = None)
+sealed trait OwnedBeforePolicyYear
+
+case object IsOwnedBefore2012 extends OwnedBeforePolicyYear
+
+case object IsOwnedBefore2017 extends OwnedBeforePolicyYear
```
Comment (daniel-manning) [inline app/models/propertyDetailsModels.scala]: These can be generic as case classes with a start year for a period.

File: app/utils/PropertyDetailsUtils.scala (hunk 3 above)
Comment (boseabhishek) [reply app/utils/PropertyDetailsUtils.scala]: Yes they could be - these are just constants used in a few places in the file..

File: app/models/propertyDetailsModels.scala (hunk 4 above)
Comment (boseabhishek) [reply app/models/propertyDetailsModels.scala]: I would like to keep them as case objects as they serve as constants and also help in exhaustive matching..

**Comment by daniel-manning** [comment]
Talking with BAs this policy period may change in the future, so it is fine to leave it as it is

File: app/models/propertyDetailsModels.scala (hunk 4 above)
Comment (daniel-manning) [reply app/models/propertyDetailsModels.scala]: yeah that comment was in the general theme of making the 5 year periods automatically and avoiding having to do that. This is fine now.

**Pull request merged**